subprojects/common/http-client/src/test/kotlin/com/avito/http/HttpClientProviderMetricsTest.kt

```diff
@@ -0,0 +1,185 @@
+package com.avito.http
+
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsMetric
+import com.avito.android.stats.StubStatsdSender
+import com.avito.android.stats.TimeMetric
+import com.avito.http.internal.RequestMetadata
+import com.avito.logger.StubLoggerFactory
+import com.avito.time.StubTimeProvider
+import com.google.common.truth.Correspondence
+import com.google.common.truth.Truth.assertThat
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.io.IOException
+import java.net.SocketTimeoutException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+internal class HttpClientProviderMetricsTest {
+
+    private val loggerFactory = StubLoggerFactory
+
+    private val timeProvider = StubTimeProvider()
+
+    private val statsDSender = StubStatsdSender()
+
+    private val mockWebServer = MockWebServer()
+
+    private val metricNamesCorrespondence: Correspondence<StatsMetric, StatsMetric> = Correspondence.from(
+        { actual, expected ->
+            if (actual != null && expected != null) {
+                actual.name == expected.name
+            } else false
+        },
+        "metric names are equal"
+    )
+
+    private val doesNotMatter = 111L
+
+    @Test
+    fun `timeout metric send - request timed out`() {
+        val provider = createClientProvider()
+
+        val httpClient = provider.provide()
+            .readTimeout(1, TimeUnit.MILLISECONDS)
+            .build()
+
+        mockWebServer.enqueue(MockResponse().setHeadersDelay(10, TimeUnit.MILLISECONDS))
+
+        assertThrows<SocketTimeoutException> {
+            httpClient.blockingCall()
+        }
+
+        assertThat(statsDSender.getSentMetrics()).comparingElementsUsing(metricNamesCorrespondence).containsExactly(
+            TimeMetric(SeriesName.create("service", "some-service", "some-method", "timeout"), doesNotMatter)
+        )
+    }
+
+    @Test
+    fun `all metrics send - multiple parallel requests`() {
```

Comment:
This test is flaky.
Please, take a look at Teamcity builds for #939 PR.

Reply:
Created MBS-11108 for this

Reply:
Thanks!